example/src/App.tsx

```diff
@@ -29,75 +28,81 @@ function App() {
 
-    const res2 = await Wallet.request('getAddresses', {
-      purposes: [AddressPurpose.Ordinals, AddressPurpose.Payment],
-    });
+      const res2 = await Wallet.request('getAddresses', {
+        purposes: [AddressPurpose.Ordinals, AddressPurpose.Payment],
+      });
 
-    if (res2.status === 'error') {
-      console.error('Error retrieving bitcoin addresses after having requested permissions.');
-      console.error(res2);
-      return;
-    }
+      if (res2.status === 'error') {
+        console.error('Error retrieving bitcoin addresses after having requested permissions.');
+        console.error(res2);
+        return;
+      }
 
-    setBtcAddressInfo(res2.result.addresses);
-    const res3 = await Wallet.request('stx_getAddresses', null);
+      setBtcAddressInfo(res2.result.addresses);
+      const res3 = await Wallet.request('stx_getAddresses', null);
 
-    if (res3.status === 'error') {
-      alert(
-        'Error retrieving stacks addresses after having requested permissions. Details in terminal.'
-      );
-      console.error(res3);
-      return;
-    }
+      if (res3.status === 'error') {
+        alert(
+          'Error retrieving stacks addresses after having requested permissions. Details in terminal.'
+        );
+        console.error(res3);
+        return;
+      }
 
-    setStxAddressInfo(res3.result.addresses);
+      setStxAddressInfo(res3.result.addresses);
+    })().catch(console.error);
   }, [setBtcAddressInfo, setStxAddressInfo]);
 
-  const onDisconnect = () => {
-    Wallet.disconnect();
-    setBtcAddressInfo([]);
-    setStxAddressInfo([]);
-  };
+  const onDisconnect = useCallback(() => {
+    (async () => {
+      await Wallet.disconnect();
+      setBtcAddressInfo([]);
+      setStxAddressInfo([]);
+    })().catch(console.error);
+  }, [setBtcAddressInfo, setStxAddressInfo]);
 
   if (!isConnected) {
     return (
-      <div className="App">
-        <header className="App-header">
-          <img className="logo" src="/sats-connect.svg" alt="SatsConnect" />
+      <Container>
+        <Header>
+          <Logo src="/sats-connect.svg" alt="SatsConnect" />
           <NetworkSelector network={network} setNetwork={setNetwork} />
           <p>Click the button to connect your wallet</p>
           <ConnectButtonsContainer>
-            <button onClick={onConnect}>Connect</button>
-            <button onClick={onConnectLegacy}>Connect (Legacy)</button>
+            <Button onClick={onConnect}>Connect</Button>
+            <Button onClick={onConnectLegacy}>Connect (Legacy)</Button>
           </ConnectButtonsContainer>
-        </header>
-      </div>
+        </Header>
+      </Container>
     );
   }
 
   return (
-    <div className="App">
-      <div className="App-body">
+    <Container>
+      <Body>
         <div>
-          <img className="logo" src="/sats-connect.svg" alt="SatsConnect" />
+          <Logo src="/sats-connect.svg" alt="SatsConnect" />
         </div>
         <AddressDisplay
```

Comment:
the legacyAddressInfo is not passed to the address display so it doesn't get displayed if you are using `getAccounts`